### PR TITLE
[Performance][Attention] Optimize PCP FA restore and output merge

### PR DESCRIPTION
**Pull request by li1how**
### What this PR does / why we need it?

This PR optimizes the PCP hybrid attention path:

- Builds and passes `pcp_fa_padding_restore_idx` so the FA QKV restore path can use `index_select` with a zero-padding row instead of boolean scatter into a workspace.
- Fixes PCP FA padding calculation for improved memory management.

These changes reduce restore overhead in PCP attention while keeping the behavior unchanged.

### Does this PR introduce _any_ user-facing change?

No. This is an internal performance optimization for PCP attention.

### How was this patch tested?

- Qwen3.5-35B-A3B-w8a8-org, A3, Tp=2, PCP=2, EP=true

- vLLM version: v0.23.0
- vLLM main: https://github.com/vllm-project/vllm/commit/1f486d96a17303ce8db8e02be39545b2be338446

### CODE REVIEW

File: vllm_ascend/attention/context_parallel/common_cp.py [hunk 1]
```diff
@@ -191,6 +192,19 @@ def _update_out_and_lse(out_list: torch.Tensor, lse_list: torch.Tensor) -> torch
...
+        out_flat = out_flat.to(torch.float32)
+    if lse_flat.dtype != torch.float32:
+        lse_flat = lse_flat.to(torch.float32)
+
+    out_items = tuple(out_flat.unbind(0))
+    lse_items = tuple(lse_flat.unbind(0))
+    out_final, _ = torch_npu.npu_attention_update(lse_items, out_items, 0)
+    out_final = out_final.view(batch_size, num_heads, head_size)
+
+    lse_final = torch.logsumexp(lse_flat, dim=0, keepdim=False)
```
Comment (realliujiaxu) [inline vllm_ascend/attention/context_parallel/common_cp.py]: please check if `dtype` of `lse_final` and `lse_final` keeps unchanged

Comment (li1how) [reply vllm_ascend/attention/context_parallel/common_cp.py]: No, lse_final does not keep the original dtype; it becomes torch.float32 because npu_attention_update requires lse to be FLOAT, and casting lse to FP32 should not reduce precision.

File: vllm_ascend/attention/context_parallel/common_cp.py [hunk 2]
```diff
@@ -191,6 +192,19 @@ def _update_out_and_lse(out_list: torch.Tensor, lse_list: torch.Tensor) -> torch
...
+    lse_final = lse_final.view(batch_size, num_heads, 1)
     return out_final, lse_final
```
[the remark this answers is not in the document]
Comment (li1how) [reply vllm_ascend/attention/context_parallel/common_cp.py]: Updated. out_final is now cast back to the original attention output dtype before returning, while lse_final remains float32 because npu_attention_update requires lse to be FLOAT.

**Review by weiguihua2 — APPROVED** [review]

**Review by realliujiaxu — APPROVED** [review]
LGTM

**Comment by li1how** [comment]
/cherry-pick releases/v0.23.0
[Bot]: cherry-pick completed successfully. New PR created: https://github.com/vllm-project/vllm-ascend/pull/11842
